tooling/templatize/pkg/config/config_test.go

```diff
@@ -28,7 +28,7 @@ func TestConfigProvider(t *testing.T) {
 	assert.Equal(t, "aro-hcp-int.azurecr.io/maestro-server:the-stable-one", variables["maestro_image"])
 
 	// key is in the config file, default, varaible value
-	assert.Equal(t, fmt.Sprintf("hcp-underlay-%s-%s", region, regionStamp), variables["region_resourcegroup"])
+	assert.Equal(t, fmt.Sprintf("hcp-underlay-%s-%s", region, stamp), variables["region_resourcegroup"])
 }
 
 func TestInterfaceToVariable(t *testing.T) {
```

Comment:
Please add a test here with nested maps in `i`.

Reply:
Please add test cases to `TestMergeVariable` for nested sub-maps overwriting parent maps (both at one level of overwriting and more than one)

Reply:
* test for nested maps for TestInterfaceToVariable done (+ fixed a bug)
* not sure about the additional `TestMergeVariable` test cases you asked about. have a look at the two onces i added